internal/policy/verify.go

```diff
@@ -329,6 +329,119 @@ func VerifyRelativeForRef(ctx context.Context, repo *gitinterface.Repository, fi
 	return nil
 }
 
+// VerifyMergeable checks if a change in a featureRef can be merged into the
+// targetRef. For Git reference protection rules, if a rule can be met with one
+// additional signature, this function returns true and without errors to
+// indicate that the merge will meet policy _if_ the resultant RSL entry is
+// signed by an approved party.
+func VerifyMergeable(ctx context.Context, repo *gitinterface.Repository, targetRef, featureRef string) (bool, error) {
+	slog.Debug("Loading policy...")
+	policyState, err := LoadCurrentState(ctx, repo, PolicyRef)
+	if err != nil {
+		return false, err
+	}
+
+	slog.Debug("Loading current set of attestations...")
+	attestationsState, err := attestations.LoadCurrentAttestations(repo)
+	if err != nil {
+		return false, err
+	}
+
+	var fromID gitinterface.Hash
+	slog.Debug(fmt.Sprintf("Identifying latest RSL entry for '%s'...", targetRef))
+	targetEntry, _, err := rsl.GetLatestReferenceEntry(repo, rsl.ForReference(targetRef), rsl.IsUnskipped())
+	switch {
+	case err == nil:
+		fromID = targetEntry.TargetID
+	case errors.Is(err, rsl.ErrRSLEntryNotFound):
+		fromID = gitinterface.ZeroHash
+	default:
+		return false, err
+	}
+
+	slog.Debug(fmt.Sprintf("Identifying latest RSL entry for '%s'...", featureRef))
+	featureEntry, _, err := rsl.GetLatestReferenceEntry(repo, rsl.ForReference(featureRef), rsl.IsUnskipped())
+	if err != nil {
+		return false, err
+	}
+
+	mergeTreeID, err := repo.GetMergeTree(fromID, featureEntry.TargetID)
+	if err != nil {
+		return false, err
+	}
+
+	authorizationAttestation, approverKeyIDs, err := getApproverAttestationAndKeyIDsForIndex(ctx, repo, policyState, attestationsState, targetRef, fromID, mergeTreeID)
+	if err != nil {
+		return false, err
+	}
+
+	verifiers, err := policyState.FindVerifiersForPath(fmt.Sprintf("%s:%s", gitReferenceRuleScheme, targetRef))
+	if err != nil {
+		return false, err
+	}
+
+	_, rslEntrySignatureNeededForThreshold, err := verifyGitObjectAndAttestationsUsingVerifiers(ctx, verifiers, gitinterface.ZeroHash, authorizationAttestation, approverKeyIDs, true)
```

Comment:
> @wlynch: Where is the gitObj signature being done? It's not immediately obvious from this code so this looks like we're checking the same thing twice.

This code has changed quite a bit since this comment was left. For clarity, I've applied it here and I think it's most relevant here. In a typical verification flow, we pass in the commit ID of the RSL entry. Here, we're passing in zero because the RSL entry doesn't exist yet. So, effectively we're saying only verify the signatures on the attestation, and we say verifyMergeable=true so that the called function knows to test for threshold-1. The idea is that if we meet threshold-1 using the attestations at present, then with a valid RSL entry we can meet threshold.

Reply:
I think it's safe to mark this as addressed.